perun-core/src/main/java/cz/metacentrum/perun/core/impl/Synchronizer.java

```diff
@@ -90,32 +93,67 @@ public void checkMembersState() {
 				vosMap.put(vo.getId(), vo);
 			}
 
+			Calendar monthBefore = Calendar.getInstance();
+			monthBefore.add(Calendar.MONTH, 1);
+
 			// log message for all members which will expire in 30 days
-			compareDate.add(Calendar.DAY_OF_MONTH, 30);
-			List<Member> expireIn30Days = perunBl.getSearcherBl().getMembersByExpiration(sess, "=", compareDate);
-			for (Member m : expireIn30Days) {
-				getPerun().getAuditer().log(sess, "{} will expire in {} days in {}.", m, 30, vosMap.get(m.getVoId()));
+
+			List<Member> expireInAMonth = perunBl.getSearcherBl().getMembersByExpiration(sess, "=", monthBefore);
+			for (Member m : expireInAMonth) {
+				try {
+					perunBl.getMembersManagerBl().canExtendMembershipWithReason(sess, m);
+					getPerun().getAuditer().log(sess, "{} will expire in a month in {}.", m, vosMap.get(m.getVoId()));
+				} catch (ExtendMembershipException ex) {
+					if (!Objects.equals(ex.getReason(), ExtendMembershipException.Reason.OUTSIDEEXTENSIONPERIOD)) {
+						// we don't care about other reasons (LoA), user can update it later
+						getPerun().getAuditer().log(sess, "{} will expire in a month in {}.", m, vosMap.get(m.getVoId()));
+					}
+				}
 			}
 
 			// log message for all members which will expire in 14 days
-			compareDate.add(Calendar.DAY_OF_MONTH, -16);
+			compareDate.add(Calendar.DAY_OF_MONTH, 14);
```

Comment:
I would prefer to use always a new variable with specific current date + or - some day. Recycling this variable is not bad, but it is harder to read and debug.